### PR TITLE
Adding a note about DNS to the generic HTTP error message

### LINKED ISSUE #257: Unhelpful error when using `yarn` when system is unable to resolve package domains via DNS
I recently had an update forced on my router which updated my DNS settings and for whatever reason it failed to resolve the host for the yarnpkg domain and, as a result, this error was thrown:

```
skaiser [~/Dev/redacted] > yarn --verbose                                                                                                                                      
Internal Error: Error when performing the request
    at ClientRequest.<anonymous> (/opt/homebrew/Cellar/corepack/0.17.2/libexec/lib/node_modules/corepack/dist/corepack.js:43953:14)
    at ClientRequest.emit (node:events:513:28)
    at TLSSocket.socketErrorListener (node:_http_client:495:9)
    at TLSSocket.emit (node:events:513:28)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

After like an hour of debugging I thought _**maybe**_ it could be related to my credentials so I tried `yarn login` and after logging in all of the issues went away. However, later the issues came back and `yarn login` was unable to resolve them... I think the first time I was having DNS issues but had yarn installed and the second time I think I was having the same issues but the installer itself couldn't be found.

Any idea what might have caused this? I'm going to try and replicate but wanted to open this issue to help prevent others from going down the same rabbit hole I did.

### PR DESCRIPTION
**Pull request by SpencerKaiser**
Fixes #257

### CODE REVIEW

File: sources/httpUtils.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export async function fetchUrlStream(url: string, options: RequestOptions = {})
     });
 
     request.on(`error`, err => {
-      reject(new Error(`Error when performing the request`));
+      reject(new Error(`Error when performing the request; this may be due to a DNS error, try \`curl ${url}\` to help diagnose the issue`));
```
[the remark this answers is not in the document]
Comment (SpencerKaiser) [reply sources/httpUtils.ts:25]: Yea I wasn't happy about this either... honestly what would likely be ideal is a permalink to a debugging markdown file here in GitHub... thoughts on that?

The reason I'm not a huge fan of _**not**_ being explicit here is that the error I had was with my ipv6 DNS, not the ipv4, so my browser could hit the page for some reason but `curl` could not...

I loathe networking stuff, so none of this is my expertise; feel free to correct/guide me as you think is relevant! Just trying to advocate for my fellow networking haters! haha

[the remark this answers is not in the document]
Comment (aduh95) [reply sources/httpUtils.ts:25]: I’d prefer we had a Troubleshooting section e.g. in the README with a detailed explanation than trying to fit all the info in error messages, would that make sense? We can also link to the readme in the error message if we think it’s necessary.

[the remark this answers is not in the document]
Comment (SpencerKaiser) [reply sources/httpUtils.ts:25]: I like that way more; I'll swap over to that

[the remark this answers is not in the document]
Comment (SpencerKaiser) [reply sources/httpUtils.ts]: Just pushed some changes and also linted that file; I was getting all sorts of eslint errors locally when using the eslint config for the repo

**Comment by SpencerKaiser** [comment]
Updated error after latest push:

```console
Internal Error: Error when performing the request to https://registry.yarnpkg.com/yarn/-/yarn-1.3.1.tgz; for troubleshooting help, see https://github.com/nodejs/corepack#troubleshooting
    at ClientRequest.<anonymous> (/Users/skaiser/Dev/corepack/dist/corepack.js:43961:9)
    at ClientRequest.emit (node:events:513:28)
    at TLSSocket.socketErrorListener (node:_http_client:502:9)
    at TLSSocket.emit (node:events:513:28)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

File: README.md [hunk 2]
```diff
@@ -217,6 +217,16 @@ network interaction.
 - `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY` are supported through
   [`node-proxy-agent`](https://github.com/TooTallNate/node-proxy-agent).
 
+## Troubleshooting
+
+### Networking
+
+There are a wide variety of networking issues that can occur while running `corepack` commands. Things to check:
+
+- Make sure your network connection is active if you are trying to retrieve remote content (such as a version of `yarn` that is not installed locally)
```
Comment (aduh95) [inline README.md:226]:
```suggestion
- Make sure your network connection is active.
```

File: README.md [hunk 3]
```diff
@@ -217,6 +217,16 @@ network interaction.
...
+- Make sure the host for your request can be resolved by your DNS; try using `curl [URL]` from your shell
```
Comment (aduh95) [inline README.md:227]:
```suggestion
- Make sure the host for your request can be resolved by your DNS; try using `curl [URL]` from your shell.
```

File: README.md [hunk 4]
```diff
@@ -217,6 +217,16 @@ network interaction.
...
+- Check your proxy settings (see [Environment Variables](#environment-variables))
```
Comment (aduh95) [inline README.md:228]:
```suggestion
- Check your proxy settings (see [Environment Variables](#environment-variables)).
```

**Review by aduh95 — COMMENTED** [review]
LGTM with a few nits. Can we remove the unrelated changes? We can take them to a separate PR if you want.

**Review by aduh95 — APPROVED** [review]

**Pull request merged**